### PR TITLE
Add documentation for MQTT Update platform

### PR DESCRIPTION
**Pull request by bieniu**
## Proposed change

This PR adds documentation for MQTT Update platform.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/80659
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/update.mqtt.markdown [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+   description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all`, `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any`, `payload_available` must be received on at least one configured availability topic before the entity is marked as online. If set to `latest`, the last `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
+   required: false
+   type: string
+   default: latest
+availability_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `availability_topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
+  required: false
+  type: template
+command_topic:
+  description: The MQTT topic to publish command to start installing process.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]: May be explain what payload will be send on this topic?

File: source/_integrations/update.mqtt.markdown [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  required: false
+  type: integer
+  default: 0
+retain:
+  description: If the published message should have the retain flag on or not.
+  required: false
+  type: boolean
+  default: false
+state_topic:
+  description: The MQTT topic subscribed to receive update of the installed version.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: The MQTT topic subscribed to receive an update of the installed version.
```

File: source/_integrations/update.mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/update.mqtt.markdown]: Refer to `payload_install` as value that is published.

File: source/_integrations/update.mqtt.markdown [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+json_attributes_topic:
+  description: The MQTT topic subscribed to receive a JSON dictionary payload and then set as entity attributes. Implies `force_update` of the current select state when a message is received on this topic.
+  required: false
+  type: string
+latest_version_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the latest version value."
+  required: false
+  type: template
+latest_version_topic:
+  description: The MQTT topic subscribed to receive update of the latest version.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: The MQTT topic subscribed to receive an update of the latest version.
```

File: source/_integrations/update.mqtt.markdown [hunk 4]
```diff
@@ -0,0 +1,213 @@
+---
+title: "MQTT Update"
+description: "Instructions on how to interact with a device exposing an Update entity through MQTT from within Home Assistant."
+ha_category:
+  - Update
+ha_release: 2021.7
```
Comment (frenck) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
ha_release: "2021.11"
```

File: source/_integrations/update.mqtt.markdown [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+ha_iot_class: Configurable
+ha_domain: mqtt
+---
+
+The `mqtt` Update platform allows you to integrate devices that might expose firmware/software installed and latest versions through MQTT into Home Assistant as an Update entity. Every time a message under the `topic` in the configuration is received, the entity will be updated in Home Assistant.
```
Comment (frenck) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
The `mqtt` Update platform allows you to integrate devices that might expose firmware/software installed and the latest versions through MQTT into Home Assistant as an Update entity. Every time a message under the `topic` in the configuration is received, the entity will be updated in Home Assistant.
```

File: source/_integrations/update.mqtt.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/update.mqtt.markdown]: @bieniu Could you take a look? Or @jbouwh Could you suggest an edit?

Comment (frenck) [reply source/_integrations/update.mqtt.markdown]: (for the remaining comments, so we can get it ready before beta lands)

File: source/_integrations/update.mqtt.markdown [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+   description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all`, `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any`, `payload_available` must be received on at least one configured availability topic before the entity is marked as online. If set to `latest`, the last `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
+   required: false
+   type: string
+   default: latest
+availability_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `availability_topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
+  required: false
+  type: template
+command_topic:
+  description: The MQTT topic to publish command to start installing process.
```
Comment (jbouwh) [inline source/_integrations/update.mqtt.markdown]:
```suggestion
  description: The MQTT topic to publish `payload_install` to start installing process.
```

**Review by frenck — APPROVED** [review]
Thanks, @bieniu & @jbouwh 👍

**Comment by bieniu** [comment]
@frenck @jbouwh Sorry guys, I was offline. And thanks!

**Pull request merged**